js/customize-snapshots.js

```diff
@@ -273,8 +299,216 @@
 		 *
 		 * @returns {string} URL.
 		 */
-		getSnapshotFrontendPreviewUrl: function() {
+		getSnapshotFrontendPreviewUrl: function getSnapshotFrontendPreviewUrl() {
 			return api.previewer.getFrontendPreviewUrl();
+		},
+
+		/**
+		 * Resets conflicts.
+		 *
+		 * @param {boolean} saved - If saved or not.
+		 * @return {void}
+		 */
+		resetConflicts: function resetConflicts( saved ) {
+			var snapshot = this;
+
+			if ( ! saved ) {
+				return;
+			}
+
+			_.each( snapshot.conflict.controlThickboxes, function( thickBox ) {
+				thickBox.remove();
+			} );
+
+			if ( snapshot.conflict.currentRequest ) {
+				snapshot.conflict.currentRequest.abort();
+				snapshot.conflict.currentRequest = null;
+			}
+
+			snapshot.conflict.controlThickboxes = {};
+			snapshot.conflict.controlsWithPendingRequest = {};
+
+			api.control.each( function( control ) {
+				_.each( control.settings, function( setting ) {
+					setting.unbind( snapshot.handleConflictRequestOnFirstChange );
+				} );
+				control.notifications.remove( snapshot.conflict.notificationCode );
+				snapshot.addConflictButton( control );
+			} );
+		},
+
+		/**
+		 * Handle conflict on first settings change.
+		 *
+		 * @param {wp.customize.Control} control - Control.
+		 * @return {void}
+		 */
+		handleConflictRequestOnFirstChange: function handleConflictRequestOnFirstChange( control ) {
+			var snapshot = this;
+
+			_.each( control.settings, function( setting ) {
+				setting.unbind( snapshot.handleConflictRequestOnFirstChange );
+				snapshot.handleConflictRequest( setting, control );
+			} );
+		},
+
+		/**
+		 * Add conflict button icon on first change and trigger handleConflictRequest.
+		 *
+		 * @param {object} control where conflict button will be added.
+		 *
+		 * @return {void}
+		 */
+		addConflictButton: function addConflictButton( control ) {
+			var snapshot = this;
+
+			control.deferred.embedded.done( function() {
+				var hasDirty, updateCurrentValue, bindFirstChange = false;
+
+				if ( ! control.setting || api.section( control.section() ).extended( api.OuterSection ) ) {
+					return;
+				}
+
+				updateCurrentValue = function() {
+					_.each( control.settings, function( setting ) {
+						snapshot.updateConflictValueMarkup( setting.id, control.container );
+					} );
+				};
+
+				hasDirty = _.find( control.settings, function( setting ) {
+					return setting._dirty;
+				} );
+
+				if ( hasDirty ) {
+					snapshot.handleConflictRequestOnFirstChange( control );
+				} else {
+					bindFirstChange = true;
+				}
+				_.each( control.settings, function( setting ) {
+					if ( bindFirstChange ) {
+						setting.bind( function() {
```

Comment:
This anonymous function is being bound here, and then there seems to be an attempt to unbind elsewhere with `setting.unbind( snapshot.handleConflictRequestOnFirstChange );` but it won't work because it's not the same function. I think a different approach to doing once is needed here, for example:

```diff
diff --git a/js/customize-snapshots.js b/js/customize-snapshots.js
index cf94a3b..0082b5b 100644
--- a/js/customize-snapshots.js
+++ b/js/customize-snapshots.js
@@ -347,7 +347,6 @@
 			var snapshot = this;
 
 			_.each( control.settings, function( setting ) {
-				setting.unbind( snapshot.handleConflictRequestOnFirstChange );
 				snapshot.handleConflictRequest( setting, control );
 			} );
 		},
@@ -386,10 +385,17 @@
 					bindFirstChange = true;
 				}
 				_.each( control.settings, function( setting ) {
+					var changeOnce;
 					if ( bindFirstChange ) {
-						setting.bind( function() {
+						changeOnce = function () {
 							snapshot.handleConflictRequestOnFirstChange( control );
-						} );
+							setting.unbind( changeOnce );
+						};
+						setting.bind( changeOnce );
 					}
 					setting.bind( updateCurrentValue );
 				} );
```

Reply:
But this can't get reset in `resetConflicts` since the function reference will not be available.